### PR TITLE
Fix (discover): Fix dismiss callout and next styling

### PR DESCRIPTION
**Pull request by joshuarrrr**
### Description

Fixes some small styling issues identified by @KrooshalUX and @kgcreative when reviewing `next` theming

1. Fix `onDismiss` prop of callout
2. Put callout in panel for better alignment in discover
3. Add `subdued` to combobox panel
4. Use `primary`-derived color for field titles in discover (and lint-ignore)

Note - I tried to fix the histogram rect highlight color in `next` dark mode, but can't figure out the problem. I got as far as confirming that both `discover` and `legacy_discover` pass the same styling props, but somehow it's not applied the same in each case.

I also notices some other minor fit and finish issues in the video:
1. Layout jank when hovering on the source, as the inline text adjusts its width to make room for the expand button
2. Loading spinner sometimes overlaps callout and other elements
3. Time range not maintained when switching
4. The way field names break across lines seems less than ideal

## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/1679762/67e4a134-ccef-4db4-b958-270abf7becfb

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by kavilla — APPROVED** [review]

**Comment by kavilla** [comment]
Backporting for 2.10 right?

**Review by abbyhu2000 — APPROVED** [review]

File: src/plugins/discover/public/application/view_components/canvas/index.tsx [hunk 1]
```diff
@@ -49,11 +49,27 @@ export default function DiscoverCanvas({ setHeaderActionMenu, history }: ViewPro
   if (isCallOutVisible) {
     callOut = (
       <EuiFlexItem grow={false}>
+        <EuiPanel hasBorder={false} hasShadow={false} color="transparent" paddingSize="s">
```
Comment (ashwin-pc) [inline src/plugins/discover/public/application/view_components/canvas/index.tsx:52]: Why do we have 2 callouts here?

**Review by ashwin-pc — DISMISSED** [review]

**Review by abbyhu2000 — DISMISSED** [review]

File: src/plugins/discover/public/application/view_components/canvas/index.tsx (hunk 1 above)
Comment (joshuarrrr) [reply src/plugins/discover/public/application/view_components/canvas/index.tsx:52]: that was a merge conflict - should be fixed

Comment (joshuarrrr) [reply src/plugins/discover/public/application/view_components/canvas/index.tsx:52]: oh, somehow managed to not commit - let me try again.

Comment (joshuarrrr) [reply src/plugins/discover/public/application/view_components/canvas/index.tsx:52]: ha, OK, yeah, this should be fixed - but doublecheck my resolution here.

**Comment by kavilla** [comment]
@joshuarrrr  looks like it just needs a snapshot upgrade

**Comment by kavilla** [comment]
Will set DCO to pass, it's being squashed anyways.

**Review by kavilla — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Review by abbyhu2000 — APPROVED** [review]
